### PR TITLE
Added Nginx example configuration

### PR DESCRIPTION
**Pull request by yuriescl**
Anchors might need a reference configuration for Nginx. It took me some time to get it working properly, specially the HTTPS and proxy.

### CODE REVIEW

**Review by JakeUrban — COMMENTED** [review]
Just to confirm my understanding, the following env variables would be used with this configuration:

- LOCAL_MODE: false
- SECURE_SSL_REDIRECT: true
- SECURE_PROXY_SSL_HEADER: ("HTTP_X_FORWARDED_PROTO", "https")

Is your `HOST_URL` `http://localhost:8000` or `stellar-anchor.mydomain.com`?

**Comment by yuriescl** [comment]
> Just to confirm my understanding, the following env variables would be used with this configuration:
> 
>     * LOCAL_MODE: false
> 
>     * SECURE_SSL_REDIRECT: true
> 
>     * SECURE_PROXY_SSL_HEADER: ("HTTP_X_FORWARDED_PROTO", "https")
> 
> 
> Is your `HOST_URL` `http://localhost:8000` or `stellar-anchor.mydomain.com`?

This is what I'm using:

```
POLARIS_LOCAL_MODE = False
SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
SECURE_SSL_REDIRECT = False  # Nginx does the redirection
```

**Comment by yuriescl** [comment]
> Is your HOST_URL http://localhost:8000 or stellar-anchor.mydomain.com?
`POLARIS_HOST_URL='https://stellar-anchor.mydomain.com'`

**Comment by yuriescl** [comment]
I have had issues with SEP-24 because Polaris was generating the interactive URL with `http://`.
I had to set this line in the Nginx config to fix that:
`proxy_redirect http://localhost:8000 https://stellar-anchor.mydomain.com`
Otherwise Polaris would give something like:
```
{
  "type": "interactive_customer_info_needed",
  "url" : "http://stellar-anchor.mydomain.com.....",
  "id": "82fhs729f63dh0v4"
}
```

**Comment by yuriescl** [comment]
https://github.com/stellar/django-polaris/issues/449 is related to this. It's a reverse proxy configuration issue, that's why I think we should have a reference Nginx config file on the repo, so Anchors can spot which configuration they might be missing.

**Comment by JakeUrban** [comment]
> I have had issues with SEP-24 because Polaris was generating the interactive URL with `http://`.
> I had to set this line in the Nginx config to fix that:
> `proxy_redirect http://localhost:8000 https://stellar-anchor.mydomain.com`
> Otherwise Polaris would give something like:
> 
> ```
> {
>   "type": "interactive_customer_info_needed",
>   "url" : "http://stellar-anchor.mydomain.com.....",
>   "id": "82fhs729f63dh0v4"
> }
> ```

Nginx's `proxy_redirect` seems to only concern itself with the `Location` and `Refresh` response headers, not an attribute within a JSON response body. I would imagine that `SECURE_PROXY_SSL_HEADER` being set informs Django to construct HTTPS URLs if the original request used HTTPS, are you sure the `url` JSON attribute uses http without the `proxy_redirect` directive?

**Comment by JakeUrban** [comment]
Looking at the [Django source code](https://github.com/django/django/blob/8bcb00858e0ddec79cc96669c238d29c30d7effb/django/http/request.py#L231), it seems `SECURE_PROXY_SSL_HEADER` is all you need.

**Comment by yuriescl** [comment]
> Looking at the [Django source code](https://github.com/django/django/blob/8bcb00858e0ddec79cc96669c238d29c30d7effb/django/http/request.py#L231), it seems `SECURE_PROXY_SSL_HEADER` is all you need.

Just tested, that is correct. My reverse proxy config is set like this and it's correctly returning `https` in the JSON url:
`settings.py`:
```
SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
SECURE_SSL_REDIRECT = False
```
`nginx.conf`:
```
    location / {
        proxy_pass http://localhost:8000;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
```
For some reason that I don't remember I had to add `proxy_redirect http://localhost:8006 https://stellar-anchor.mydomain.com` to Nginx. It's probably not required.

File: example/nginx-letsencrypt.conf [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Example configuration for Django Polaris using Nginx as reverse proxy with HTTPS enabled.
```
Comment (JakeUrban) [inline example/nginx-letsencrypt.conf:1]: Can you add a comment specifying the linux distribution used? The only reason this is necessary is due to the file-system paths generated by Lets Encrypt, which may not be the correct paths on other linux distributions.

Lets also mention that this is not used in the deployment of this example application, but is purely for reference.

**Review by JakeUrban — COMMENTED** [review]
A couple more requests to add some comments, but after they're made I'll approve 👍

File: example/nginx-letsencrypt.conf (hunk 1 above)
Comment (yuriescl) [reply example/nginx-letsencrypt.conf]: Done, thanks

**Review by JakeUrban — APPROVED** [review]

**Pull request merged**